platform/swift/source/FatalIssueReporter.swift

```diff
@@ -0,0 +1,180 @@
+              let config = IssueReporterConfig.from(contents)
+        else {
+            return .failure(.malformedConfigFile)
+        }
+
+        do {
+            try FileManager.default.createDirectory(at: destDir, withIntermediateDirectories: true)
+        } catch {
+            return .failure(.processingFailure("failed to create upload directory: \(error)"))
+        }
+
+        return .success((config, destDir))
+    }
+
+    /// Ensure a URL entry is a valid report by checking it is a regular file with a predetermined file extension
+    ///
+    /// - parameter entry:         a URL
+    /// - parameter fileExtension: expected file extension
+    ///
+    /// - returns: true if a given entry is a regular file and ends with `fileExtension`
+    private static func isReportURL(_ entry: URL, fileExtension: String) -> Bool {
+        if let properties = try? entry.resourceValues(forKeys: [.nameKey, .isRegularFileKey]),
+           let name = properties.name,
+           let isFile = properties.isRegularFile {
+            return isFile && name.hasSuffix(".\(fileExtension)")
+        }
+        return false
+    }
+
+    /// Copy the first file matching the config to the destination directory
+    ///
+    /// - parameter config:  base directory and file extension to use
+    /// - parameter destDir: copy destination
+    ///
+    /// - returns: initialization resolution indicating whether a file was copied or not
+    private static func copyFiles(config: IssueReporterConfig, destDir: URL) -> IssueReporterInitState {
+        let resourceNames: [URLResourceKey] = [.nameKey, .isRegularFileKey, .contentModificationDateKey]
+        guard let sourceDir = getCacheDir()?.appendingPathComponent(config.rootDir),
+              let files = try? FileManager.default.contentsOfDirectory(at: sourceDir, includingPropertiesForKeys: resourceNames),
+              let entry = files.filter({ entry in
+                return isReportURL(entry, fileExtension: config.fileExtension)
+              }).first
+        else {
+            return .initialized(.withoutPriorCrash)
+        }
+        let dest = destDir.appendingPathComponent(getUploadFilename(entry))
+        do {
+            try FileManager.default.copyItem(at: entry, to: dest)
+            return .initialized(.sent)
+        } catch {
+            return .initialized(.processingFailure("failed to copy report: \(error)"))
+        }
+    }
+
+    /// Create upload filename for a given URL
+    ///
+    /// - parameter entry: a file URL
+    ///
+    /// - returns: a new file name for `entry`, prepending the modification date as a Unix timestamp
+    private static func getUploadFilename(_ entry: URL) -> String {
+        if let properties = try? entry.resourceValues(forKeys: [.contentModificationDateKey]),
+           let modDate = properties.contentModificationDate {
+            let modDateMillis = UInt64(modDate.timeIntervalSince1970 * 1000)
```

Comment:
does this mean we actually don't have the precision of ms and we're just adding 0s?

Reply:
@snowp just clarified that this is actually returning fractions originally so we should be fine